### PR TITLE
[Task-66][feat][refactor] Task 66 forgot password rework

### PR DESCRIPTION
**Pull request by OlegPanasyuk**
Updated links for requests and processing of responses.

### CODE REVIEW

File: src/services/auth.js [hunk 1]
```diff
@@ -62,38 +62,30 @@ export const changePassword = async (id, data) => {
...
   try {
     const response = await httpService.post({ url: links.resetRoute, data });
-    return response;
+    return response.data;
   } catch (e) {
     return {
       status: 400,
-      data: 'Bad email',
+      message: 'Bad email',
     };
```
Comment (slavasgod) [inline src/services/auth.js]: You throwing error from the backend and don't catch it here and just hardcoded the error message. Look how we catch errors on `loginRequest`.

File: src/containers/auth/forgotPassword/index.js [hunk 2]
```diff
@@ -75,9 +75,9 @@ class ForgotPassword extends PureComponent {
     const valid = this.preValidateForm({ email });
     if (valid) {
       const response = await forgotPasswordRequest({ email });
-      const { status, data } = response;
-      const shownMessage = status < 300 ? 'successShow' : 'errorShow';
-      this.showCurrentMessage({ keys: [shownMessage], message: data });
+      const { status, message } = response;
+      const shownMessage = status < 300 ? ['successShow'] : ['formShow', 'validationShow'];
+      this.showCurrentMessage({ keys: shownMessage, message });
     }
```
Comment (slavasgod) [inline src/containers/auth/forgotPassword/index.js:49]: We already have a modal component, so wrap your component on it.

I don't like showMessage component and how you tied with statuses. As I understood, you use it only in your component, why you can't just use component state and display these messages depends on this state

**Review by slavasgod — CHANGES_REQUESTED** [review]

File: src/containers/auth/forgotPassword/index.js [hunk 3]
```diff
@@ -75,9 +75,9 @@ class ForgotPassword extends PureComponent {
     const valid = this.preValidateForm({ email });
     if (valid) {
       const response = await forgotPasswordRequest({ email });
```
Comment (universezzz) [inline src/containers/auth/forgotPassword/index.js]: You can write like this` const { status, message } = await forgotPasswordRequest({ email });`

File: src/services/auth.js (hunk 1 above)
Comment (OlegPanasyuk) [reply src/services/auth.js]: Done

File: src/containers/auth/forgotPassword/index.js (hunk 3 above)
Comment (OlegPanasyuk) [reply src/containers/auth/forgotPassword/index.js]: Done

File: src/containers/auth/forgotPassword/index.js (hunk 2 above)
Comment (OlegPanasyuk) [reply src/containers/auth/forgotPassword/index.js:49]: Done

**Review by slavasgod — APPROVED** [review]

File: src/containers/auth/resetPassword/styles.module.scss [hunk 4]
```diff
@@ -45,11 +46,11 @@
...
 
-  .btn_right_position {
+  .btnRightPosition {
     align-self: flex-end;
   }
 
-  .btn_margin_top {
+  .btnMarginTop {
     margin-top: 25px;
   }
```
Comment (melonholy) [inline src/containers/auth/resetPassword/styles.module.scss]: Can we unite this in one style and rename it? i mean not btn

File: src/containers/auth/resetPassword/index.js [hunk 5]
```diff
@@ -3,202 +3,141 @@ import React, { PureComponent } from 'react';
...
+        {pending && (
+          <div className={styles.form}>
+            <Spinner loading={pending} />
+          </div>
+        )}
+
+        {!pending && success && (
+          <>
+            <div
+              className={`${styles.form} ${styles.successMessage} ${styles.bigSizeMessage}`}>
```
Comment (melonholy) [inline src/containers/auth/resetPassword/index.js]: also unite it

**Review by melonholy — CHANGES_REQUESTED** [review]

File: src/containers/auth/resetPassword/styles.module.scss (hunk 4 above)
Comment (OlegPanasyuk) [reply src/containers/auth/resetPassword/styles.module.scss]: Done

File: src/containers/auth/resetPassword/index.js (hunk 5 above)
Comment (OlegPanasyuk) [reply src/containers/auth/resetPassword/index.js]: This step will add repeating of code.

Comment (OlegPanasyuk) [reply src/containers/auth/resetPassword/index.js]: Done

File: src/containers/auth/resetPassword/styles.module.scss [hunk 6]
```diff
@@ -17,44 +19,43 @@
...
+    align-items: center;
     box-sizing: border-box;
   }
 
   .row {
     margin: 15px 0 0 0;
   }
 
-  .row_top {
+  .rowTop {
```
Comment (melonholy) [inline src/containers/auth/resetPassword/styles.module.scss]: Remove it

**Review by melonholy — APPROVED** [review]

**Pull request merged**